osu.Game.Rulesets.Osu/Skinning/Legacy/LegacyCursorParticles.cs

```diff
@@ -84,6 +105,50 @@ protected override void Update()
                 kiaiHitObject = playfield.HitObjectContainer.AliveObjects.FirstOrDefault(isTracking);
 
             kiaiSpewer.Active.Value = kiaiHitObject != null;
+
+            if (relaxing != relaxHitSpewer.Active.Value) relaxHitSpewer.Active.Value = relaxing;
+
+            if (relaxingKeyPressed() && !kiaiSpewer.Active.Value)
+            {
+                relaxHitSpewer.Direction = SpewDirection.None;
+                relaxHitSpewer.Colour = starBreakAdditive;
+                relaxHitSpewer.SpawnKiaiParticles = true;
+            }
+            else
+            {
+                relaxHitSpewer.SpawnKiaiParticles = false;
+            }
+        }
+
+        private bool doRelaxingSpew()
+        {
+            if (playfield == null || gameplayState == null) return false;
+
+            bool hitting = playfield.HitObjectContainer.AliveObjects.FirstOrDefault(isHit) != null;
+            bool slidingOrSpinning = playfield.HitObjectContainer.AliveObjects.FirstOrDefault(isTracking) != null;
+
+            return hitting && relaxingKeyPressed() && !breakSpewer.Active.Value && !slidingOrSpinning;
+        }
+
+        private bool relaxingKeyPressed()
+        {
+            if (gameplayState == null) return false;
+
+            return relaxing && (leftPressed || rightPressed);
+        }
+
+        private bool isHit(DrawableHitObject h)
+        {
+            switch (h)
+            {
+                case DrawableSlider slider:
+                    return slider.HeadCircle.IsHit;
+
+                case DrawableHitCircle hitCircle:
+                    return hitCircle.IsHit;
+            }
+
+            return false;
```

Comment:
These should probably be one method, because this logic is really stretched across too much